rfc/src/rfcs/0012-list.md

```diff
@@ -0,0 +1,174 @@
+
+This feature will not cause any regressions for exisiting users.
+
+## User Experience
+
+Users run a `list` subcommand, which prints metadata about the harnesses and contracts in each crate under verification. The subcommand will take the option `--message-format=[human|json]`, which changes the output format. The default is `human`, which prints to the terminal. The `json` option creates and writes to a JSON file instead.
+
+This subcommand will not fail. In the case that it does not find any harnesses or contracts, it will print a message informing the user of that fact.
+
+### Human Format
+
+The default format, `human`, will print the harnesses and contracts in a project, along with the total counts of each.
+
+For example:
+
+```
+Kani Version: 0.5.4
+
+Standard Harnesses:
+- example::verify::check_new
+- example::verify::check_modify
+
+Contract Harnesses:
+- example::verify::check_foo_u32
+- example::verify::check_foo_u64
+- example::verify::check_func
+- example::verify::check_bar
+
+Contracts:
+|--------------------------|-----------------------------------------------|
+| Function                 |  Contract Harnesses                           |
+|--------------------------|-----------------------------------------------|
+| example::impl::func      |  example::verify::check_func                  |
+|--------------------------|-----------------------------------------------|
+| example::impl::bar       |  example::verify::check_bar                   |
+|--------------------------|-----------------------------------------------|
+| example::impl::foo       |  example::verify::check_foo_u32,              |
+|                          |  example::verify::check_foo_u64               |
+|--------------------------|-----------------------------------------------|
+| example::prep::parse     |  NONE                                         |
+|--------------------------|-----------------------------------------------|
+
+Totals:
+- Standard Harnesses: 2
+- Contract Harnesses: 4
+- Functions with Contracts: 4
+- Contracts: 10
+```
+
+A "Standard Harness" is a `#[proof]` harness, while a "Contract Harness" is a `#[proof_for_contract]` harness.
+
+### JSON Format
+
+As the name implies, the goal of the `human` output is to be friendly for human readers. If the user wants an output format that's more easily parsed by a script, they can use the `json` option. 
+
+The JSON format will contain the same information as the human format, with the addition of file paths and JSON version. The JSON version number will start at zero and increment whenever we make an update to the format. This way, any users relying on this format for their scripts can realize that changes have occurred and update their logic accordingly.
+
+For example:
+
+```json
+{
+    kani-version: 0.5.4,
+    json-version: 0,
```

Comment:
maybe just `version` or `file-version`. `json-version` sounds like the `json` version.